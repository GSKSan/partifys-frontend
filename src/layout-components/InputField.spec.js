import React from 'react'
import renderer from 'react-test-renderer'
import InputField from './InputField'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'

const theme = {
  colors: {
    dark: '#151515',
    danger: '#a63d40',
    primary: '#1db954',
    confirm: '#90a959',
    accent: '#6494aa',
    typography: '#373d3f',
    typographyLight: '#bcbcbc'
  }
}

describe('InputField', () => {
  it('Should render an empty text InputField by default', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <InputField
            className='test-textfield'
            label='My label'
            name='test'
            value=''
            placeholder='write something'
            onChange={jest.fn()}
          />
        </ThemeProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Should render an empty Textfield', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <InputField
            className='test-textfield'
            type='email'
            label='Email address'
            name='email'
            value=''
            placeholder='john.doe@company.com'
            onChange={jest.fn()}
          />
        </ThemeProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Should render a filled InputField', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <InputField
            className='test-textfield'
            type='email'
            label='Email address'
            name='email'
            value='jane.doe@company.com'
            placeholder='john.doe@company.com'
            onChange={jest.fn()}
          />
        </ThemeProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})