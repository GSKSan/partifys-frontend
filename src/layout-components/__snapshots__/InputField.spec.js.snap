// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Should render a filled Textfield 1`] = `
<input
  className="inputLogin iconKey"
  name="email"
  placeholder="john.doe@company.com"
  type="email"
/>
`;

exports[`TextField Should render an empty Textfield 1`] = `
<input
  className="inputLogin iconKey"
  name="email"
  placeholder="john.doe@company.com"
  type="email"
/>
`;

exports[`TextField Should render an empty text TextField by default 1`] = `
<input
  className="inputLogin iconKey"
  name="test"
  placeholder="write something"
/>
`;
